elements/lisk-codec/src/string.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+import { writeVarInt } from './varint';
+
+interface SchemaProperty {
+	readonly dataType: string;
+}
+
+export const writeString = (value: string, _schema: SchemaProperty): Buffer =>
+	Buffer.concat([
+		writeVarInt(Buffer.from(value, 'utf8').length, { dataType: 'uint32' }),
+		Buffer.from(value, 'utf8'),
+	]);
```

Comment:
```suggestion
{
const buffer = Buffer.from(value, 'utf8');
Buffer.concat([
writeVarInt(buffer.length, {dataType: 'uint32'}),
buffer
])
return buffer
```

Reply:
This needs to be checked as I didn't run it but this way you only copy the string into the buffer once and then just read the length of that buffer

Reply:
I think you meant 
```typescript
export const writeString = (value: string, _schema: SchemaProperty): Buffer => {
	const stringBuffer = Buffer.from(value, 'utf8');

	return Buffer.concat([
		writeVarInt(stringBuffer.length, { dataType: 'uint32' }),
		stringBuffer,
	]);
};
```

Reply:
I think yes? You're not using the extra var I think here right?

Reply:
I'm using `stringBuffer` here to create the buffer to be returned as you suggested except that you were returning the `stringBuffer` itself in the writer, I'm returning the concatenated buffer in the end.

Reply:
hard to write in github comment box :)